opcua/server/history_sql.py

```diff
@@ -244,24 +246,22 @@ def read_event_history(self, source_id, start, end, nb_values, evfilter):
             # select events from the database; SQL select clause is built from EventFilter and available fields
             try:
                 for row in _c_read.execute('SELECT "_Timestamp", {cl} FROM "{tn}" WHERE "_Timestamp" BETWEEN ? AND ? '
-                                           'ORDER BY "_Id" {dir} LIMIT ?'.format(cl=clauses, tn=table, dir=order),
+                                           'ORDER BY "_Id" {dir} LIMIT ?'.format(cl=clauses_str, tn=table, dir=order),
                                            (start_time, end_time, limit,)):
 
-                    # place all the variants in the event field list object
-                    hist_ev_field_list = ua.HistoryEventFieldList()
-                    i = 0
-                    for field in row:
+                    fdict = {}
+                    for i, field in enumerate(row):
                         # if the field is the _Timestamp column store it in a list used for getting the continuation
                         if i == 0:
                             cont_timestamps.append(field)
                         else:
                             if field is not None:
-                                hist_ev_field_list.EventFields.append(ua.Variant.from_binary(Buffer(field)))
+                                fdict[clauses[i+1]] = ua.Variant.from_binary(Buffer(field))
+                                print(clauses[i+1], fdict[clauses[i+1]])
```

Comment:
here we should just fill the dict with name:variant. but I do not know how to get name, or I am missing EventType